[Bugfix][Benchmarks] Fix a benchmark of deepspeed-mii backend to use api_key

tishizaki opened the pull request:
Previously, a benchmark can't run to a deepspeed-mii server with using api_key.

This patch is ported the implementation from
async_request_openai_completions() so that api_key can be specified.

<details>
<summary>Bug reproduction</summary>

server command:
```text
$ python -m mii.entrypoints.openai_api_server --tensor-parallel 1 --model meta-llama/Meta-Llama-3-8B --port 8000 --api-keys testabc
```

client command:
```text
$ export OPENAI_API_KEY=testabc
$ python vllm/benchmarks/benchmark_serving.py --dataset-name sharegpt --dataset-path ./ShareGPT_V3_unfiltered_cleaned_split.json --model meta-llama/Meta-Llama-3-8B --num_prompts 500 --request-rate 4 --port 8000 --backend deepspeed-mii
```

</details>

ywang96 commented on benchmarks/backend_request_func.py:
```diff
@@ -197,6 +197,11 @@ async def async_request_deepspeed_mii(
     request_func_input: RequestFuncInput,
     pbar: Optional[tqdm] = None,
 ) -> RequestFuncOutput:
+    api_url = request_func_input.api_url
+    assert api_url.endswith(
+        ("completions", "profile")
+    ), "OpenAI Completions API URL must end with 'completions' or 'profile'."
```
I couldn't find any documentation where it says deepspeed-mii supports OpenAI compatible API. https://github.com/deepspeedai/DeepSpeed-MII?tab=readme-ov-file#restful-api

Could you point to where it says deepspeed-mii is OpenAI compatible?

tishizaki replied:
I think this patch added to support OpenAI compatible API.
https://github.com/deepspeedai/DeepSpeed-MII/pull/317

ywang96 reviewed (commented):
Thanks for the fix - if deepspeed-mii supports OpenAI compatible API, is there a reason why we can't simply point it to `async_request_openai_completions` and delete `async_request_deepspeed_mii`? The latter was added by me back then when it didn't support OpenAI API.

tishizaki commented:
@ywang96 Thanks for your reply.

At first, I also thought it would be possible to integrate it into async_request_openai_completions. However, I thought it would be better to integrate it after considering the differences in payload parameters (especially temperature) and making streaming compatible (ttft measurement).
I think that supporting api_key is the first step.

tishizaki committed:
Fix a benchmark of deepspeed-mii backend to use api_key

Previously, a benchmark can't run to a deepspeed-mii server
with using api_key.

This patch is ported the implementation from
async_request_openai_completions() so that api_key can be specified.

Signed-off-by: Teruaki Ishizaki <teruaki.ishizaki@ntt.com>
